### PR TITLE
CD-174: XClarity (Lenovo) Processors Health

### PR DESCRIPTION
**Pull request by mariemm97**
… JSON

### CODE REVIEW

File: examples/ssh/xclarity_processors_health.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "processor",
+        "health",
+        "status",
+        "clock-speed"
+    ],
+    "sample_period": 1800,
```
Comment (kgskgs) [inline examples/ssh/xclarity_processors_health.json:23]: change to sample_period_s

File: examples/ssh/xclarity_processors_health.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "version": "1.0.0",
+    "name": "XClarity Processors Health",
+    "description": "Monitor processors health on Lenovo XClarity server",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "SSH Enabled",
+            "The keyboard_interactive option can be true or false depending on the XClarity version"
```
Comment (kgskgs) [inline examples/ssh/xclarity_processors_health.json:12]: provides no information

File: examples/ssh/xclarity_processors_health.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ */
+function validate() {
+    executeCommand(command)
+        .then(parseValidateOutput)
+        .then(D.success)
+        .catch(checkSshError);
+}
+
+function parseValidateOutput(output) {
+    if (output.trim() !== "") {
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_processors_health.js:72]: Please use a proper validation on the command

File: examples/ssh/xclarity_processors_health.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+// Parse and insert processors health data into the custom table
+function parseData(output) {
+    var lines = output.split("\n");
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    for (var i = 2; i < lines.length-1; i++) {
+        var line = lines[i].trim();
+        var values = line.split(/\s+/);
+        var recordId = (values[0] + "-" + values[1]).replace(recordIdSanitisationRegex, '').slice(0, 50);
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_processors_health.js:96]: also lower case

File: examples/ssh/xclarity_processors_health.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+    "logo": "lenovo-xclarity.svg",
+    "version": "1.0.0",
+    "name": "XClarity Processors Health",
+    "description": "Monitor processors health on Lenovo XClarity server",
+    "category": "Software Integration",
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_processors_health.json:6]: change to hardware

File: examples/ssh/xclarity_processors_health.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "processor",
+        "health",
+        "status",
+        "clock-speed"
+    ],
+    "sample_period_s": 1800,
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_processors_health.json:22]: change to 30 min

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**